reflect.go

```diff
@@ -1018,3 +1014,34 @@ func writeReflectJSON(rv reflect.Value, rt reflect.Type, opts Options, w io.Writ
 	}
 
 }
+
+func isEmpty(rt reflect.Type, rv reflect.Value, opts Options) bool {
```

Comment:
This works, but I think we can do better by calling writeReflectJSON onto a buffer and seeing if the resulting buffer was an empty struct "{}" or not.

Basically, for non-empty structs that aren't comparable, this implementation is a bit slower.  Can we write a note on this for future reference?